packages/packages/v1-adapters/src/dispatchers/dispatchers.ts

```diff
@@ -0,0 +1,38 @@
+import { ExtendedWindow } from './types';
+import { isJQueryDeferred, promisifyJQueryDeferred } from './utils';
+
+export function runCommand( command: string, args?: object ) : Promise<unknown> {
+	const extendedWindow = window as unknown as ExtendedWindow;
+
+	if ( ! extendedWindow.$e?.run ) {
+		return Promise.reject( '`$e.run()` is not available' );
+	}
+
+	const result = extendedWindow.$e.run( command, args );
+
+	if ( result instanceof Promise ) {
+		return result;
+	}
+
+	if ( isJQueryDeferred( result ) ) {
+		return promisifyJQueryDeferred( result );
+	}
+
+	return Promise.resolve( result );
+}
+
+export function openRoute( route: string ) : Promise<void> {
```

Comment:
if the function always returns a Promise, why not:
```
export function openRoute( route: string ) {
	return new Promise( ... );
}
```

Reply:
Tried to write it with this approach. but having multiple conditions feels cleaner and more readable IMO